app/src/main/java/de/xikolo/storages/RecentCoursesStorage.kt

```diff
@@ -0,0 +1,51 @@
+package de.xikolo.storages
+
+import android.content.Context
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import de.xikolo.storages.base.BaseStorage
+
+typealias RecentCourse = Pair<String, String>
+
+const val MAX_SHORTCUTS = 4
+
+class RecentCoursesStorage : BaseStorage(PREF_RECENT_COURSES, Context.MODE_PRIVATE) {
+
+    companion object {
+        private const val PREF_RECENT_COURSES = "preference_recent_courses"
+
+        fun addCourseToRecentCourses(
```

Comment:
why this indirection with supplying `courses` as a parameter? Isn't it always `recentCourses`?

Reply:
This allows easier testing of the addition of new courses. Otherwise the tests have to be completely rewritten because App.Instance would have to be mocked. Should it nevertheless be changed?